packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -46,6 +46,10 @@ BuildApp() {
     target_path="${FLUTTER_TARGET}"
   fi
 
+  # Write User-Defined FLTAssetsPath to AppFrameworkInfo.plist
+  FLTAssetsPath="${FLTAssetsPath:-flutter_assets}"
+  /usr/libexec/PlistBuddy -c "Delete :FLTAssetsPath" Flutter/AppFrameworkInfo.plist
+  /usr/libexec/PlistBuddy -c "Add :FLTAssetsPath string ${FLTAssetsPath}" Flutter/AppFrameworkInfo.plist
```

Comment:
What if we did something like:

```
local assets_path=$(/usr/libexec/PlistBuddy -c "Print FLTAssetsPath" $derived_dir/AppFrameworkInfo.plist)
if [[ $? != 0 ]]; then
  assets_path="flutter_assets"
fi
```

(which would require moving the `derived_dir` logic on line 94 a bit further up in this function)

Rather than trying to modify the file, which seems a bit more risky.

It'd also be really nice if we had some way to test this, for example building a Flutter project where this value was set int he plist.

Reply:
Modifying AppFrameworkInfo.plist is necessary because [FlutterDartProject.mm#L208](https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm#L208) needs to read FLTAssetsPath from it.  
I have done optimization, please take a look.

Reply:
I'm suggesting that instead of modifying it in the script, you look to see if the user modified it.

This should be controlled by the Info.plist, not by the xcode_backend.sh